docs/MetricsSpecification.md

```diff
@@ -0,0 +1,164 @@
+  unit: "{timeout}", description: "The number of connection timeouts that have occurred trying to obtain a connection from the pool.");
+static readonly Histogram<float> CreateTimeHistogram = Meter.CreateHistogram<float>("db.client.connections.create_time",
+  unit: "ms", description: "The time it took to create a new connection.");
+static readonly Histogram<float> WaitTimeHistogram = Meter.CreateHistogram<float>("db.client.connections.wait_time",
+  unit: "ms", description: "The time it took to obtain an open connection from the pool.");
+static readonly Histogram<float> UseTimeHistogram = Meter.CreateHistogram<float>("db.client.connections.use_time",
+  unit: "ms", description: "The time it took to create a new connection.");
+```
+
+* _TODO: Use `float` as the unit for time measurement (in milliseconds), or `int` or `double`?_
+
+ADO.NET Providers SHOULD support as many of these instruments as is possible.
+Providers MUST use the names, units, and descriptions given in the example code above.
+
+For the `create_time` and `wait_time` instruments, the reported time SHOULD be as close as possible to the entire duration
+of the `DbConnection.Open(Async)` or `DbDataSource.OpenConnection(Async)` method,
+i.e., from when user code starts opening a connection to when an open `DbConnection` is returned.
+
+For the `use_time` instrument, the reported time SHOULD measure the duration of the time from when an open
+`DbConnection` is returned from `DbConnection.Open(Async)` or `DbDataSource.OpenConnection(Async)` to when
+user code closes the connection (and returns it to the pool) via `DbConnection.Dispose(Async)` or `DbConnection.Close(Async)`.
+
+If the ADO.NET Provider supports “Minimum Pool Size” and “Maximum Pool Size” connection string options and has a
+fixed pool size, then `db.client.connections.idle.min` should be set (once) to the value of “Minimum Pool Size”,
+and `db.client.connections.idle.max` and `db.client.connections.max` SHOULD each be set (once) to the value of
+“Maximum Pool Size”.
+If the ADO.NET Provider has a connection pool whose limits change dynamically during the lifetime of the application,
+then the values of these instruments MUST be updated when the pool size limits change.
+
+## Reporting Metrics
+
+Use the [`UpDownCounter<T>.Add`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.updowncounter-1.add)
+or [`Histogram<T>.Record`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.histogram-1.record)
+methods to report metrics for the instruments described above.
+
+## Example
+
+A highly simplified example of reporting metrics might look like this:
+
+```csharp
+public class XyzsqlConnection
+{
+  // Assume the existence of a connection pooling implementation.
+  private XyzsqlConnectionPool _pool;
+
+  public override async Task OpenAsync(CancellationToken cancellationToken)
```

Comment:
A similar CloseAsync example could be useful as well (e.g. to show decrementing the "usage" counter), though obviously this is all for illustration purposes anyway.